pdns/dnsdistdist/docs/rules-actions.rst

```diff
@@ -169,12 +169,62 @@ For Rules related to the incoming query:
 
   Remove all current rules.
 
-.. function:: getAction(n) -> Action
+.. function:: getAction(n) -> DNSDistRuleAction
 
-  Returns the Action associated with rule ``n``.
+  Returns the :class:`DNSDistRuleAction` associated with rule ``n``.
 
   :param int n: The rule number
 
+.. function:: getCacheHitResponseRule(selector) -> DNSDistResponseRuleAction
+
+  .. versionadded:: 1.9.0
+
+  Return the cache-hit response rule corresponding to the selector, if any.
+  The selector can be the position of the rule in the list, as an integer,
+  its name as a string or its UUID as a string as well.
+
+  :param int or str selector: The position in the list, name or UUID of the rule to return.
+
+.. function:: getCacheInsertedResponseRule(selector) -> DNSDistResponseRuleAction
+
+  .. versionadded:: 1.9.0
+
+  Return the cache-hit response rule corresponding to the selector, if any.
+  The selector can be the position of the rule in the list, as an integer,
+  its name as a string or its UUID as a string as well.
+
+  :param int or str selector: The position in the list, name or UUID of the rule to return.
+
+.. function:: getResponseRule(selector) -> DNSDistResponseRuleAction
+
+  .. versionadded:: 1.9.0
+
+  Return the response rule corresponding to the selector, if any.
+  The selector can be the position of the rule in the list, as an integer,
+  its name as a string or its UUID as a string as well.
+
+  :param int or str selector: The position in the list, name or UUID of the rule to return.
+
+.. function:: getRule(selector) -> DNSDistRuleAction
+
+  .. versionadded:: 1.9.0
+
+  Return the rule corresponding to the selector, if any.
+  The selector can be the position of the rule in the list, as an integer,
+  its name as a string or its UUID as a string as well.
+
+  :param int or str selector: The position in the list, name or UUID of the rule to return.
+
+.. function:: getSelfAnsweredResponseRule(selector) -> DNSDistResponseRuleAction
```

Comment:
This name is a bit confusing to me. "SelfAnswered" can be more than just cache-hits, right? i.e. answers synthesized by dnsdist itself. Do they count or not?

Reply:
I agree the name is not great, but it is already used by several other functions (`addSelfAnsweredResponseAction`, `mvSelfAnsweredResponseRule`, `mvSelfAnsweredResponseRuleToTop`, `mvSelfAnsweredResponseRuleToTop`, `showSelfAnsweredResponseRules`, ...). The rules are applied to all responses that comes from dnsdist and not from a backend, so cache hits and answers synthesized from rules do indeed count.

Reply:
OK, then the explanation of the count should mention that it's more than just cache-hits.

Reply:
Oh, I see now that there was a copy/paste mistake, fixed!